### ISSUE TITLE
Problems validating TLS CA Cert

### ISSUE THREAD

**Issue by rlangfordBV** [flynn/flynn#4372]
Hello,

To start, I refreshed the TLS cert on a cluster of `v20180320.0`, as it was expiring.  When validating that the process was successful, we would see the following:
```
[me@<some host> ~]$ echo | openssl s_client -showcerts -servername status.<my.cluster.url> -connect status.<my.cluster.url>:443
CONNECTED(00000003)
depth=0 C = US, ST = Texas, L = Austin, O = "<My Company>, Inc.", OU = Flynn, CN = *.<my.cluster.url>
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 C = US, ST = Texas, L = Austin, O = "<My Company>, Inc.", OU = Flynn, CN = *.<my.cluster.url>
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:/C=US/ST=Texas/L=Austin/O=<My Company>, Inc./OU=Flynn/CN=*.<my.cluster.url>
   i:/C=US/O=DigiCert Inc/CN=DigiCert SHA2 Secure Server CA
-----BEGIN CERTIFICATE-----
< REDACTED FOR BREVITY>
-----END CERTIFICATE-----
---
Server certificate
subject=/C=US/ST=Texas/L=Austin/O=<My Company>, Inc./OU=Flynn/CN=*.<my.cluster.url>
issuer=/C=US/O=DigiCert Inc/CN=DigiCert SHA2 Secure Server CA
---
No client certificate CA names sent
Peer signing digest: SHA384
Server Temp Key: ECDH, P-256, 256 bits
---
SSL handshake has read 2298 bytes and written 475 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES128-GCM-SHA256
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : ECDHE-RSA-AES128-GCM-SHA256
    Session-ID: 1DF974B3F2E1CE8CE477C755F5B4D8D020CB87EF401836C337754376E8381E48
    Session-ID-ctx:
    Master-Key: <REDACTED>
    Key-Arg   : None
    Krb5 Principal: None
    PSK identity: None
    PSK identity hint: None
    TLS session ticket:
    0000 - d7 89 4e df e5 c5 1f 5d-9f c7 cc e3 2c af 1e e4   ..N....]....,...
    0010 - 61 be 5a 61 71 87 8c db-c3 2f 7f f2 ee c8 cf b7   a.Zaq..../......
    0020 - fa ef 76 0d 96 ac 4a eb-39 7c dd 51 f7 14 f9 b2   ..v...J.9|.Q....
    0030 - a3 d9 de c3 f1 03 ee 93-27 4a dd e7 97 74 ce 2e   ........'J...t..
    0040 - 66 ed 7d f6 02 84 ae ea-2c 43 79 80 5d 1e e5 d3   f.}.....,Cy.]...
    0050 - ee 54 09 e7 9c 9a d7 72-bd a9 bd b5 2c 73 b9 dd   .T.....r....,s..
    0060 - b5 82 73 2d e7 a3 80 c4-bc 72 61 b0 00 71 e3 5f   ..s-.....ra..q._
    0070 - e0 fe 4e 42 36 be c6 3d-                          ..NB6..=

    Start Time: 1524182366
    Timeout   : 300 (sec)
    Verify return code: 21 (unable to verify the first certificate)
---
DONE
```

I published the credentials to our s3 bucket, and we started new clusters from scratch.  When trying to provision the new cluster, we got:
```
+ add_local_flynn_config
+ set +o xtrace
flynn cluster add my-cluster-cluster-timestamp my-cluster.mycompany.com $CONTROLLER_KEY
Error writing CA certificate: Get https://controller.my-cluster.mycompany.com/ca-cert: x509: certificate signed by unknown authority
+ sleep 60
```

We then created new clusters using `v20170719.0`, and using the same keys files (we store them in s3 and download them during provisioning), the clusters came up without issue.

Today, I tried to spin up clusters with `v20180420.0`, and I'm seeing the same issue I see when trying to create a cluster with `v20180320.0`.
```
+ add_local_flynn_config
+ set +o xtrace
flynn cluster add my-cluster-cluster-timestamp my-cluster.mycompany.com $CONTROLLER_KEY
Error writing CA certificate: Get https://controller.my-cluster.mycompany.com/ca-cert: x509: certificate signed by unknown authority
+ sleep 60
```

Note:
* Curling `https://controller.my-cluster.mycompany.com/ca-cert`, and validating the cert using `openssl x509 -in ./digicert.crt -text -noout` is successful
```
 8 ➜  ~  openssl x509 -in ./digicert.crt -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            01:fd:a3:eb:6e:ca:75:c8:88:43:8b:72:4b:cf:bc:91
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert Global Root CA
        Validity
            Not Before: Mar  8 12:00:00 2013 GMT
            Not After : Mar  8 12:00:00 2023 GMT
        Subject: C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
            RSA Public Key: (2048 bit)
                Modulus (2048 bit):
                    <REDACTED>
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Basic Constraints: critical
                CA:TRUE, pathlen:0
            X509v3 Key Usage: critical
                Digital Signature, Certificate Sign, CRL Sign
            Authority Information Access:
                OCSP - URI:http://ocsp.digicert.com

            X509v3 CRL Distribution Points:
                URI:http://crl3.digicert.com/DigiCertGlobalRootCA.crl
                URI:http://crl4.digicert.com/DigiCertGlobalRootCA.crl

            X509v3 Certificate Policies:
                Policy: X509v3 Any Policy
                  CPS: https://www.digicert.com/CPS

            X509v3 Subject Key Identifier:
                0F:80:61:1C:82:31:61:D5:2F:28:E7:8D:46:38:B4:2C:E1:C6:D9:E2
            X509v3 Authority Key Identifier:
                keyid:03:DE:50:35:56:D1:4C:BB:66:F0:A3:E2:1B:1B:C3:97:B2:3D:D1:55

    Signature Algorithm: sha256WithRSAEncryption
        <REDACTED>
 8 ➜  ~
```
* Cluster with `v20170719.0` works, versions `v20180320.0` and `v20180420.0` do not work.  All clusters are using the same key material stored in an s3 bucket
* Cert is issued by DigiCert, so we're using a known good CA

What, if anything has changed w.r.t. cert handling between `v20170719.0` and `v20180320.0`?

**Comment by rlangfordBV** [comment]
Noticed some new nightly builds have been released, and tested `v20180501.0`, which seems to work.

**Comment by titanous** [comment]
Flynn is unmaintained and our infrastructure will shut down on June 1, 2021. See [the README](https://github.com/flynn/flynn#flynn-is-unmaintained) for details.
